### PR TITLE
add: Keptn CLI auto completion docs

### LINKED ISSUE #729: Provide docs for auto-completion feature in CLI
Similar to https://kubernetes.io/docs/tasks/tools/install-kubectl/#enabling-shell-autocompletion we need a documentation on how to use the auto-completion feature in the CLI.

Currently, it is not clear how to use it, as e.g., this issue shows: https://github.com/keptn/keptn/issues/3212

### PR DESCRIPTION
**Pull request by DarthBenro008**
- Fixes #729
 
Preview: 
docs_1

### CODE REVIEW

File: content/docs/0.8.x/reference/cli/_index.md [hunk 1]
```diff
@@ -198,3 +198,144 @@ The `KEPTNCONFIG` environment variable holds a config file that contains the cre
...
+```
+
+After reloading your shell, Keptn CLI autocompletion will be enabled succesfully.
+</details>
+
+<details><summary>Autocompletion for Bash (MacOS)</summary>
+
+The Keptn CLI completion script for Bash can be generated with `keptn completion bash`. Sourcing this script in your shell enables Keptn CLI autocompletion.
+
+However, the Keptb CLI autocompletion script depends on bash-completion which you thus have to previously install.
```
Comment (christian-kreuzberger-dtx) [inline content/docs/0.8.x/reference/cli/_index.md:258]: Typo *Keptb* -> *Keptn*

File: content/docs/0.8.x/reference/cli/_index.md [hunk 2]
```diff
@@ -198,3 +198,144 @@ The `KEPTNCONFIG` environment variable holds a config file that contains the cre
...
+
+Keptn CLI provides autocompletion support for Bash and Zsh, which can save you a lot of typing.
+
+Below are the procedures to set up autocompletion for Bash and Zsh. Please expand the corresponding section matching your configuration.
+
+<details><summary>Autocompletion for Bash (Linux)</summary>
+
+The Keptn CLI completion script for Bash can be generated with the command `keptn completion bash`. Sourcing the completion script in your shell enables Keptn CLI autocompletion.
+
+However, the completion script depends on bash-completion, which means that you have to install this software first (you can test if you have bash-completion already installed by running type _init_completion).
```
Comment (christian-kreuzberger-dtx) [inline content/docs/0.8.x/reference/cli/_index.md:212]: Please change type _init_completion to `type _init_completion` (add single quotes)

File: content/docs/0.8.x/reference/cli/_index.md [hunk 3]
```diff
@@ -198,3 +198,144 @@ The `KEPTNCONFIG` environment variable holds a config file that contains the cre
...
+
+However, the Keptb CLI autocompletion script depends on bash-completion which you thus have to previously install.
+
+> Warning: There are two versions of bash-completion, v1 and v2. V1 is for Bash 3.2 (which is the default on macOS), and v2 is for Bash 4.1+. The Keptn CLI autocompletion script doesn't work correctly with bash-completion v1 and Bash 3.2. It requires bash-completion v2 and Bash 4.1+. Thus, to be able to correctly use the Keptn CLI autocompletion on macOS, you have to install and use Bash 4.1+ (instructions). The following instructions assume that you use Bash 4.1+ (that is, any Bash version of 4.1 or newer).
+
+### Install bash-completion for MacOS
+
+Note: As mentioned, these instructions assume you use Bash 4.1+, which means you will install bash-completion v2 (in contrast to Bash 3.2 and bash-completion v1, in which case Keptn CLI completion won't work).
+
+You can test if you have bash-completion v2 already installed with type _init_completion. If not, you can install it with Homebrew:
```
Comment (christian-kreuzberger-dtx) [inline content/docs/0.8.x/reference/cli/_index.md:266]: Same as above, please change type _init_completion to `type _init_completion` (add single quotes)

**Review by christian-kreuzberger-dtx — CHANGES_REQUESTED** [review]
Thanks! I've left some comments. Also, please remove the files from the  resources/_gen folder (I don't know why they keep re-generating, we certainly should add them to .gitignore and generate them within the CI, but that's for another day).

File: content/docs/0.8.x/reference/cli/_index.md (hunk 1 above)
Comment (DarthBenro008) [reply content/docs/0.8.x/reference/cli/_index.md]: Apologies, missed this while proofreading 😅

**Comment by DarthBenro008** [comment]
@christian-kreuzberger-dtx I have made the necessary changes. Regarding the auto-generated files, I would really like to open up an issue regarding the same and work on it!

**Comment by christian-kreuzberger-dtx** [comment]
Closing this in favor of https://github.com/keptn/keptn.github.io/pull/760

**Pull request closed without merging**